File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int right = 0;
        for(int i=0; i<piles.length; i++){
            right = Math.max(right, piles[i]);
        }
        int left = 0;
        int res = right;

        while(left <= right){
            int k = (left+right)/2;
            int hours = 0;
            for(int p=0; p<piles.length; p++){
                hours += Math.ceil((double)piles[p]/k);
            }
            if(hours <= h){
                res= Math.min(res, k);
                right = k-1;
            }else{
                left = k+1;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0
        def dfs(node, curr_h):
            if not node:
                return 
            if curr_h <= node.val:
                self.res += 1
            curr_h = max(curr_h, node.val)
            dfs(node.left, curr_h)
            dfs(node.right, curr_h)
        dfs(root, -101)
        return self.res